pkg/bootstrap/config.go

```diff
@@ -79,15 +79,28 @@ func (cfg Config) toTemplateParams() (map[string]interface{}, error) {
 		xdsType = "DELTA_GRPC"
 	}
 
+	// Add the meshScope
+	meshScope := "default"
+	if scope, found := cfg.Metadata.Labels["gke.io/mesh"]; found {
+		meshScope = scope
+	}
+
 	opts = append(opts,
 		option.NodeID(cfg.ID),
 		option.NodeType(cfg.ID),
 		option.PilotSubjectAltName(cfg.Metadata.PilotSubjectAltName),
 		option.OutlierLogPath(cfg.Metadata.OutlierLogPath),
 		option.ProvCert(cfg.Metadata.ProvCert),
 		option.DiscoveryHost(discHost),
+		option.GCPMeshScope(meshScope),
 		option.XdsType(xdsType))
 
+	// Add GCPProjectNumber to access in bootstrap template.
+	md := cfg.Metadata.PlatformMetadata
+	if projectNumber, found := md[platform.GCPProjectNumber]; found {
+		opts = append(opts, option.GCPProjectNumber(projectNumber))
```

Comment:
maybe should set it in the same place project name is? https://github.com/istio/istio/blob/0a5a486d1c69ea6c8bf96a22dc1b8daf254cf646/pkg/bootstrap/config.go#L99

Reply:
I wanted this to be unconditional, but I can put it there if you think that is the right place.

Reply:
Do you set STSPort in your case? I would think so but not certain.

It does seem weird to have ID in one place and Name in another

Reply:
I tried moving it below, and it failed for me on default configuration and deployment.

Reply:
Curious, how does it fail exactly?

Reply:
The nodeId does not find the ProjectName as the logic to populate does not get executed.

Reply:
Ok so sts port is indeed not always used. Thanks!